### ISSUE TITLE
rule/require-array-items: false positive

### ISSUE THREAD

**Issue by icholy** [Redocly/redocly-cli-cookbook#30]
Adding a `type` to the `bar` property makes the error go away, but I'm not clear on how the `where` is matching that schema in the first place.

swagger.json:

``` json
{
	"swagger": "2.0",
	"definitions": {
		"Foo": {
			"type": "object",
			"properties": {
				"bar": {}
			}
		}
	}
}
```
redocly.yml

https://github.com/Redocly/redocly-cli-cookbook/tree/main/configurable-rules/required-items-for-array-schemas

``` yml
apis:
  main:
    root: swagger.json

rules:
  rule/require-array-items:
    subject:
      type: Schema
    assertions:
      required:
        - items
    where:
      - subject:
          type: Schema
          property: type
        assertions:
          const: array
    message: The 'items' field is required for schemas of array type.
```

Error:

```
$ npx redocly lint swagger.json
validating swagger.json...
[1] swagger.json:7:5 at #/definitions/Foo/properties/bar

The 'items' field is required for schemas of array type.

5 | 			"type": "object",
6 | 			"properties": {
7 | 				"bar": {}
8 | 			}
9 | 		}

Error was generated by the rule/require-array-items rule.


swagger.json: validated in 6ms

❌ Validation failed with 1 error.
run `redocly lint --generate-ignore-file` to add all problems to the ignore file.
```

**Comment by adamaltman** [comment]
Your rule is saying that every schema type has to be an array. I think you may mean to make the rule different such as "if it is an array, then it needs to declare a list of items" (or something like that)?

**Comment by icholy** [comment]
@adamaltman I didn't write this rule, it's from this repo: https://github.com/Redocly/redocly-cli-cookbook/tree/main/configurable-rules/required-items-for-array-schemas

That being said, I've annotated the example with comments providing my understanding of each part of the rule:

``` yaml
rules:
  rule/require-array-items:
    # schema types
    subject:
      type: Schema
    # require and items property
    assertions:
      required:
        - items
    # if their type=array
    where:
      - subject:
          type: Schema
          property: type
        assertions:
          const: array
    message: The 'items' field is required for schemas of array type.
```

Can you explain where my reasoning is off?

**Comment by tatomyr** [comment]
@icholy interesting. Really it shouldn't. Will take a look at this when I have some time.

**Comment by icholy** [comment]
@tatomyr thanks for the quick turn-around! Should I open a PR or do you have it covered?

**Comment by lornajane** [comment]
Fixed by #31
